File: Arrays/287-Find-the-Duplicate-Number/JavaScript/287-Find-the-Duplicate-Number-Brute Force.js
/**
 * Problem: Merge Sorted Arrays
 * Approach: Brute Force
 *
 * Combine both arrays into one array, sort it,
 * then distribute the elements back into arr1 and arr2.
 *
 * Time Complexity: O((n + m) log(n + m))
 * Space Complexity: O(n + m)
 */

/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {void}
 */

var mergeArrays = function(arr1, arr2) {
    const temp = [...arr1, ...arr2];

    temp.sort((a, b) => a - b);

    const n = arr1.length;

    for (let i = 0; i < n; i++) {
        arr1[i] = temp[i];
    }

    for (let i = 0; i < arr2.length; i++) {
        arr2[i] = temp[n + i];
    }
};

File: Arrays/287-Find-the-Duplicate-Number/Python/287-Find-the-Duplicate-Number-Brute Force.py
"""
Problem: 287. Find the Duplicate Number
Approach: Brute Force - Sorting

Sort the array so that duplicate values become adjacent.
Return the first pair of equal adjacent elements.

Time Complexity: O(n log n)
Space Complexity: O(1) auxiliary
"""


class Solution:
    def findDuplicate(self, nums: list[int]) -> int:
        nums.sort()

        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                return nums[i]

        return -1